### PR TITLE
Monorepo troubles doc

### PR DESCRIPTION
**Pull request by DaedalusG**
**This is a (very) rough draft** for a PR intended to give folks with monorepos an idea about why monorepos cause specific problems. Its very much in a draft state and I think it could use a better look particularly in terms of the code links. Sharing now to get some insights from the search-core and codeintel teams about the validity of the information contained here (especially the code links 😅 . Hoping this can be instructive to customers and team members alike till we can rollout improvements search-core is working on. 

TL;DR features that default to unindexed symbols search when a zoekt index search is considered stale are likely to default because monorepos usually receive frequent commits. In these cases, gitserver and symbols try to lazily respond and this can result in timeouts or long waits for results. Our current best answer for this is please give gitserver and symbols more CPU.

I believe I've identified the most common frustrations here, but the codeintel one is a little dubious.

### CODE REVIEW

**Review by chrismwendt — APPROVED** [review]
The code links LGTM 👍 and I can't think of any other tips that customers can try to improve the perf aside from adding more CPUs (symbols scales linearly with CPUs, whereas adding more replicas won't help much), so I think you've covered the topic well.

I'm planning to implement incremental indexing of symbols, which I think will significantly improve these pain points. https://github.com/sourcegraph/sourcegraph/issues/27611

**Review by airamare01 — APPROVED** [review]
Looks great! Thanks @DaedalusG !
